components/style/properties/gecko.mako.rs

```diff
@@ -1028,6 +1028,75 @@ fn static_assert() {
+        use gecko_bindings::structs::nsStyleImageLayers_Size_Dimension;
+        use gecko_bindings::structs::nsStyleImageLayers_Size_DimensionType;
+        use gecko_bindings::structs::{nsStyleCoord_CalcValue, nsStyleImageLayers_Size};
+        use properties::longhands::background_size::single_value::computed_value::T;
+        use values::computed::LengthOrPercentageOrAuto;
+
+        let mut width = nsStyleCoord_CalcValue::new();
+        let mut height = nsStyleCoord_CalcValue::new();
+
+        let (w_type, h_type) = match servo {
+            T::Explicit(size) => {
+                let mut w_type = nsStyleImageLayers_Size_DimensionType::eAuto;
+                let mut h_type = nsStyleImageLayers_Size_DimensionType::eAuto;
+                if let Some(w) = size.width.to_calc_value() {
+                    width = w;
+                    w_type = nsStyleImageLayers_Size_DimensionType::eLengthPercentage;
+                }
+                if let Some(h) = size.height.to_calc_value() {
+                    height = h;
+                    h_type = nsStyleImageLayers_Size_DimensionType::eLengthPercentage;
+                }
+                (w_type, h_type)
+            }
+            T::Cover => (nsStyleImageLayers_Size_DimensionType::eCover,
+                         nsStyleImageLayers_Size_DimensionType::eCover),
+            T::Contain => (nsStyleImageLayers_Size_DimensionType::eContain,
+                         nsStyleImageLayers_Size_DimensionType::eContain),
+        };
+
+        nsStyleImageLayers_Size {
+            mWidth: nsStyleImageLayers_Size_Dimension { _base: width },
+            mHeight: nsStyleImageLayers_Size_Dimension { _base: height },
+            mWidthType: w_type as u8,
+            mHeightType: h_type as u8,
+        }
+    </%self:simple_background_array_property>
+
+    pub fn clone_background_size(&self) -> longhands::background_size::computed_value::T {
+        use gecko_bindings::structs::nsStyleCoord_CalcValue as CalcValue;
+        use gecko_bindings::structs::nsStyleImageLayers_Size_DimensionType as DimensionType;
+        use properties::longhands::background_size::single_value::computed_value::{ExplicitSize, T};
+        use values::computed::LengthOrPercentageOrAuto;
+
+        fn to_servo(value: CalcValue, ty: u8) -> LengthOrPercentageOrAuto {
+            if ty == DimensionType::eAuto as u8 {
+                LengthOrPercentageOrAuto::Auto
+            } else {
+                LengthOrPercentageOrAuto::Calc(value.into())
+            }
+        }
+
+        longhands::background_size::computed_value::T(
+            self.gecko.mImage.mLayers.iter().map(|ref layer| {
+                if DimensionType::eCover as u8 == layer.mSize.mWidthType {
+                    return T::Cover
+                }
+                if DimensionType::eContain as u8 == layer.mSize.mWidthType {
+                    return T::Contain
+                }
+
```

Comment:
Do we want to `assert!` here what `DimensionType` we expect too? (Not totally sure if we can).